packages/hint-manifest-icons/src/hint.ts

```diff
@@ -0,0 +1,150 @@
+import {
+    ManifestEvents,
+    ManifestParsed,
+    ManifestImageResource
+} from '@hint/parser-manifest';
+import { debug as d } from '@hint/utils';
+
+import meta from './meta';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class ManifestIconHint implements IHint {
+    public static readonly meta = meta;
+    public constructor(context: HintContext<ManifestEvents>) {
+        /**
+         * Try to see if the `icon` file actually
+         * exists and is accessible.
+         */
+        const iconExists = async (IconPath: string) => {
+            let networkData: NetworkData;
+
+            try {
+                networkData = await context.fetchContent(IconPath);
+            } catch (e) {
+                debug(`Failed to fetch the ${IconPath} file`);
+                const message = `'${IconPath}' could not be fetched (request failed).`;
+
+                context.report(IconPath, message);
+
+                return;
+            }
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                const message = `'${IconPath}' could not be fetched (status code: ${
+                    response.statusCode
+                }).`;
+
+                context.report(IconPath, message);
+
+                return;
+            }
+
+            let image;
+
+            try {
+                image = getImageData(response.body.rawContent);
+            } catch (e) {
+                if (e instanceof TypeError) {
+                    const message = `'${IconPath}' should be a valid PNG image.`;
+
+                    context.report(IconPath, message);
+                } else {
+                    debug(`'getImageData' failed for '${IconPath}'`);
+                }
+
+                return;
+            }
+
+            // Check if the image is a PNG.
+            if (image.type !== 'png') {
+                const message = `'${IconPath}' should be a PNG image but found be a
+                                    ${image.type} image`;
+
+                context.report(IconPath, message);
+            }
+
+            // Check if the image is 180x180px.
+            /*
+             * if (image.width !== 180 || image.height !== 180) {
+             *     const message = `'${IconPath}' should be 180x180px.`;
+             */
+
+            /*
+             *     context.report(IconPath, message);
+             * }
+             */
+        };
+
+        /**
+         *
+         * @param icons array of the icons properties
+         * @param hostnameWithProtocol
+         */
+        const validIconsURLs = (
+            icons: ManifestImageResource[],
+            hostnameWithProtocol: string
+        ) => {
+            icons.forEach(async (iconProps: { src: string }) => {
```

Comment:
Don't think you need to put the type
```suggestion
            icons.forEach(async (iconProps) => {
```

Reply:
Removed.